packages/grid/_modules_/grid/GridComponentProps.ts

```diff
@@ -41,4 +47,12 @@ export interface GridComponentProps extends GridOptionsProp {
    * An error that will turn the grid into its error state and display the error component.
    */
   error?: any;
+  /**
+   * Set the whole state of the grid.
+   */
+  state?: Partial<GridState>;
```

Comment:
If I understand correctly this is the API that the people will use. My question is - the grid state is quite big and it will probably continue to grow. If we do it like this that means that people will be able to change everything in the state. My question is should we allow that? Are there cases where we don't want people to access/change certain parts of the state? Maybe we can consider adding multiple APIs that only change parts of the state (or maybe we can leave this decision for later), what do you think @dtassone @oliviertassinari ?

Reply:
We already have all the other api functions that change the state